packages/lodestar-validator/src/services/attestation.ts

```diff
@@ -177,10 +177,14 @@ export class AttestationService {
     if (duty.isAggregator) {
       const timeout = setTimeout(async (signal = abortSignal) => {
         this.logger.debug("AttestationService: Start waitForAggregate");
-        signal?.addEventListener("abort", () => {
-          clearTimeout(timeout);
-          this.logger.debug("AttestationService: Abort waitForAggregate");
-        });
+        signal?.addEventListener(
+          "abort",
+          () => {
+            clearTimeout(timeout);
+            this.logger.debug("AttestationService: Abort waitForAggregate");
+          },
+          {once: true}
+        );
```

Comment:
The abortable setTimeout should be extracted into a utility function and unit tested

Reply:
done

Reply:
er, well, actually, forgot to add the unit test.  will do that here in a min